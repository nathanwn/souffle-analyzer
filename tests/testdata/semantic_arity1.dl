.decl foo(x: number, y: number)

foo().
foo(1).
foo(1, 2).
foo(1, 2, 3).

.decl bar(x: number, y: number)

bar(x, y, z) :- foo(x, y).
