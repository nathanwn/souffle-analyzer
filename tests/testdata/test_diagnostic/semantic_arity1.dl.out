semantic_arity1.dl
0 | .decl foo(x: number, y: number)
1 |
2 | foo().
3 | foo(1).
4 | foo(1, 2).
5 | foo(1, 2, 3).
6 |
7 | .decl bar(x: number, y: number)
8 |
9 | bar(x, y, z) :- foo(x, y).

=====

-- Cursor range --
semantic_arity1.dl
2 | foo().
  | ~~~~~
-- Range --
semantic_arity1.dl
2 | foo().
  | ~~~~~
-- Message --
Number of arguments: have 0, want 2.
_____

-- Cursor range --
semantic_arity1.dl
3 | foo(1).
  | ~~~~~~
-- Range --
semantic_arity1.dl
3 | foo(1).
  | ~~~~~~
-- Message --
Number of arguments: have 1, want 2.
_____

-- Cursor range --
semantic_arity1.dl
5 | foo(1, 2, 3).
  | ~~~~~~~~~~~~
-- Range --
semantic_arity1.dl
5 | foo(1, 2, 3).
  | ~~~~~~~~~~~~
-- Message --
Number of arguments: have 3, want 2.
_____

-- Cursor range --
semantic_arity1.dl
9 | bar(x, y, z) :- foo(x, y).
  | ~~~~~~~~~~~~
-- Range --
semantic_arity1.dl
9 | bar(x, y, z) :- foo(x, y).
  | ~~~~~~~~~~~~
-- Message --
Number of arguments: have 3, want 2.
_____
